tests/core/test_wmi.py

```diff
@@ -357,12 +379,131 @@ def test_wql_filtering(self):
 
         # Check `_format_filter` logic
         no_filters = []
-        filters = [{'Name': "SomeName"}, {'Id': "SomeId"}]
+        filters = [{'Name': "SomeName", 'Id': "SomeId"}]
 
         self.assertEquals("", format_filter(no_filters))
-        self.assertEquals(" WHERE Id = 'SomeId' AND Name = 'SomeName'",
+        self.assertEquals(" WHERE ( Name = 'SomeName' AND Id = 'SomeId' )",
                           format_filter(filters))
 
+    def test_wql_multiquery_filtering(self):
+        """
+        Format the filters with multiple properties per instance to a comprehensive WQL `WHERE` clause.
+        """
+        from checks.libs.wmi import sampler
+        format_filter = sampler.WMISampler._format_filter
+
+        # Check `_format_filter` logic
+        no_filters = []
+        filters = [{'Name': "SomeName", 'Property1': "foo"}, {'Name': "OtherName", 'Property1': "bar"}]
+
+        self.assertEquals("", format_filter(no_filters))
+        self.assertEquals(" WHERE ( Property1 = 'bar' AND Name = 'OtherName' ) OR"
+                          " ( Property1 = 'foo' AND Name = 'SomeName' )",
+                          format_filter(filters))
+
+    def test_wql_empty_list(self):
+        """
+        Format filters to a comprehensive WQL `WHERE` clause skipping empty lists.
+        """
+
+        from checks.libs.wmi import sampler
+        format_filter = sampler.WMISampler._format_filter
+
+        filters = []
+        query = {}
+        query['User'] = ('=', 'luser')
+        query['SourceName'] = ('=', 'MSSQL')
+        query['EventCode'] = []
+        query['SomethingEmpty'] = []
+        query['MoreNothing'] = []
+
+        filters.append(query)
+
+        self.assertEquals(" WHERE ( SourceName = 'MSSQL' AND User = 'luser' )",
+                          format_filter(filters))
+
+    def test_wql_filtering_op_adv(self):
+        """
+        Format the filters to a comprehensive WQL `WHERE` clause w/ mixed filter containing regular and operator modified properties.
+        """
+        from checks.libs.wmi import sampler
+        format_filter = sampler.WMISampler._format_filter
+
+        # Check `_format_filter` logic
+        filters = [{'Name': "Foo%"}, {'Name': "Bar*", 'Id': ('>=', "SomeId")}, {'Name': "Zulu"}]
```

Comment:
The wildcard character in WQL `WHERE` clauses is only `%` (not `*`), I've confirmed that by reading through https://technet.microsoft.com/en-us/library/ee176998.aspx and manually testing on a windows VM.

In terms of BC, the only check that was supporting the wildcard was the `wmi_check` (see https://github.com/DataDog/dd-agent/blob/5.5.x/checks.d/wmi_check.py#L51-L55), it was actually looking for a `*` in the filters and replacing it with a `%`.

Given that the wildcard feature was broken in 5.6.x I think we could simply only allow `%` as a wildcard in the `filters`. What do you think?